### ISSUE TITLE
Style/EmptyCaseCondition removes method call taking case as argument

### ISSUE THREAD

**Issue by DMA57361** [rubocop-hq/rubocop#6511]
I'm running corrections over some older code that hasn't been looked after by Rubocop before, and I've just had the Style/EmptyCaseCondition cop auto-correct a case statement that was being passed as an argument into a method call in such a way that the method call was removed.

## Expected behavior

Change a unnecessary case being passed into a method:
```ruby
  do_some_work case
               when object.nil?
                 Object.new
               else
                object
               end
```

Into an if statement, maintaining the call to `do_some_work`

## Actual behavior

The call to `do_some_work` is unexpectedly remove.

## Steps to reproduce the problem

Given this contrived `eg.rb` file:

```ruby
def some_bad_code(object)
  do_some_work case
               when object.nil?
                 Object.new
               else
                object
               end
end

```

Run Rubocop corrections:

```
$ bundle exec rubocop --safe-auto-correct eg.rb
Inspecting 1 file
C

Offenses:

eg.rb:1:1: C: [Corrected] Style/FrozenStringLiteralComment: Missing magic comment # frozen_string_literal: true.
def some_bad_code(object)
^
eg.rb:2:16: C: [Corrected] Style/EmptyCaseCondition: Do not use empty case condition, instead use an if expression.
  do_some_work case
               ^^^^
eg.rb:4:1: C: [Corrected] Layout/IndentationWidth: Use 2 (not 15) spaces for indentation.
               if object.nil?
^^^^^^^^^^^^^^^
eg.rb:6:16: C: [Corrected] Layout/IndentationWidth: Use 2 (not 1) spaces for indentation.
                object
               ^

1 file inspected, 4 offenses detected, 4 offenses corrected
```

The `eg.rb` file is left in this state:

```ruby
# frozen_string_literal: true

def some_bad_code(object)
  if object.nil?
    Object.new
  else
    object
  end
end
```

## RuboCop version

```
$ bundle exec rubocop -V
0.60.0 (using Parser 2.5.3.0, running on ruby 2.3.7 x86_64-darwin18)
```

**Comment by koic** [comment]
Thanks for the feedback. This issue is a regression by #6257 💦 I opened a PR #6513. Thank you!
